app/models/hotel.rb

```diff
@@ -1,3 +1,12 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+
+  validates :name, presence: true
+  validates :location, presence: true
+
+  def guest_list 
+    rooms.joins({guest_rooms: :guest}).select("guests.name").distinct
+  end
```

Comment:
Just taking a look at what you're doing here, and that's getting the names of all the guests of a specific hotel, I think this could be simplified.

```ruby
def guest_list
  rooms.joins(:guests).select("guests.name").distinct
end
```

Another much easier way to do this would be to add an additional association.

```ruby
has_many :rooms
has_many :guests, through: :rooms

def guest_list
  guests.distinct.pluck(:name)
end
```

PS - The syntax for plucking the name may be slightly off. I'm writing this without testing it. The idea remains the same.